### PR TITLE
Skus Unit Test update

### PR DESCRIPTION
**Pull request by uppalk1**
Modified few tests to remove dependencies

### CODE REVIEW

**Comment by uppalk1** [comment]
test this please !

**Comment by benbp** [comment]
@uppalk1 Are we losing test coverage for the regenerateSkus function? Is there some way to keep that or do you find it's not necessary or is implicitly tested by some of the new tests you added?

**Comment by uppalk1** [comment]
@benbp regenerate skus is being tested through some APIs. I found couple of other APIs that weren't tested, so I'll try to cover them.

File: spec/lib/api/2.0/skus-spec.js [hunk 1]
```diff
@@ -3,234 +3,238 @@
...
+    afterEach(function() {
+        skuApiService.getSkus.restore();
+        skuApiService.postSku.restore();
+        skuApiService.getSkusById.restore();
+        skuApiService.upsertSku.restore();
+        skuApiService.patchSku.restore();
+        skuApiService.regenerateSkus.restore();
+        skuApiService.getNodesSkusById.restore();
+        skuApiService.deleteSkuById.restore();
+        workflowApiService.createAndRunGraph.restore();
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: A recommendation: Use  the `sinon.sandbox` will save the effort to restore these stubs one by one.

File: spec/lib/api/2.0/skus-spec.js [hunk 2]
```diff
@@ -3,234 +3,238 @@
...
-            .expect(200, _.merge({}, sku, { packInfo: { description: null, version: null } }) );
-        });
-
-        it('should 409 reposting the same sku', function() {
-            return helper.request().post('/api/2.0/skus')
-            .send(input)
-            .expect(409);
+            .expect(200)
+            .then(function (res) {
+                    var checkSku = res.body;
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: incorrect indent

File: spec/lib/api/2.0/skus-spec.js [hunk 3]
```diff
@@ -3,234 +3,238 @@
...
         it('should 200 reputting the same sku', function() {
+            skuApiService.upsertSku.resolves(input);
             return helper.request().put('/api/2.0/skus')
             .send(input)
             .expect(200);
         });
 
         describe('PATCH /skus/:id', function () {
             var updated;
+            var updated_input = {
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: Please fix this

File: spec/lib/api/2.0/skus-spec.js [hunk 4]
```diff
@@ -3,234 +3,238 @@
...
-                    .to.have.been.calledWith({
-                        name: 'Graph.GenerateSku',
-                        options: {
-                            defaults: {
-                                nodeId: node.id
-                            }
-                        }
-                    });
-            });
+                    .expect(200)
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: Fix this please

File: spec/lib/api/2.0/skus-spec.js [hunk 5]
```diff
@@ -3,234 +3,238 @@
...
+            })
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: Fix this please

File: spec/lib/api/2.0/skus-spec.js [hunk 6]
```diff
@@ -3,234 +3,238 @@
...
-                .send({})
-                .expect(404);
-            });
-
-            it('should 404 with DELETE /skus/:id', function () {
-                return helper.request().delete('/api/2.0/skus/' + sku.id)
-                .expect(404);
+                skuApiService.getSkusById.rejects(new Errors.NotFoundError('Not Found'));
+                return helper.request().get('/api/2.0/skus/0987')
+                    .expect(404)
```
Comment (yyscamper) [inline spec/lib/api/2.0/skus-spec.js]: Fix this please

**Comment by yyscamper** [comment]
test this please

**Comment by keedya** [comment]
:+1:

**Comment by tannoa2** [comment]
+1

File: spec/lib/services/sku-pack-service-spec.js [hunk 7]
```diff
@@ -183,18 +187,154 @@ describe("SKU Pack Service", function() {
...
                 expect(val.username).equal(sku.username);
                 expect(val.password).equal(sku.password);
                 expect(val.hostname).equal(sku.hostname);
             });
         });
+
+        it('should return err status if postSku fails a non-409', function(){
+            var err = new Errors.BaseError('internal server error');
+            err.status = 500; //Checking if not 409
+            return skuService.upsertSku(sku).catch(function (err) {
```
Comment (yyscamper) [inline spec/lib/services/sku-pack-service-spec.js]: You need some special handling if you want to put  some assertion logic in the `catch` block. Image the `skuService.upsertSku` is actually resolved, so the `catch` block will not be executed. Mocha will mark the empty resolved promise as passed.  Below is some code snippet:
```javascript
return skuService.upsertSku(sku)
.then(function() {
    throw new Error('upsertSku should be rejected!');
})
.catch(function (err) {
    expect(err.status).equal(500);
});
```

File: spec/lib/services/sku-pack-service-spec.js [hunk 8]
```diff
@@ -183,18 +187,154 @@ describe("SKU Pack Service", function() {
...
+              expect(val.password).equal(sku.password);
+              expect(val.hostname).equal(sku.hostname);
+          });
+        });
+
+        it('should return an error Duplicate name found', function() {
+            var sku =
+            {"name": "my test sku"};
+            waterline.skus.findOne.resolves({});
+            return skuService.postSku(sku).catch(function(err){
```
Comment (yyscamper) [inline spec/lib/services/sku-pack-service-spec.js]: Same as my last comment, need some special code to prevent the `postSku` from being resolved.

**Comment by yyscamper** [comment]
:+1: after fix the defect if putting assertion logic into the `catch` block.

File: spec/lib/api/2.0/skus-spec.js [hunk 9]
```diff
@@ -3,241 +3,263 @@
...
-        helper.setupInjector([
-            helper.require("/lib/services/sku-pack-service"),
-            dihelper.simpleWrapper(function() { arguments[1](); }, 'rimraf')
-        ]);
         return helper.startServer([
-            dihelper.simpleWrapper(workflowApiService, 'Http.Services.Api.Workflows')
         ]).then(function() {
-            _ = helper.injector.get('_');
+            waterline = helper.injector.get('Services.Waterline');
+            sinon.stub(waterline.nodes);
```
Comment (brianparry) [inline spec/lib/api/2.0/skus-spec.js]: These stubs should be restored in your after callback so that subsequent tests do not use them.

**Pull request merged**